testcases/OpTestHMIHandling.py

```diff
@@ -212,18 +229,88 @@ def clearGardEntries(self):
                 "Failed to clear GARD entries")
             log.debug(self.cv_FSP.fspc.run_command("gard --gc cpu"))
         else:
-            g = self.cv_HOST.host_run_command("PATH=/usr/local/sbin:$PATH opal-gard list all", console=1)
-            if "No GARD entries to display" not in g:
-                self.cv_HOST.host_run_command("PATH=/usr/local/sbin:$PATH opal-gard clear all", console=1)
-                cleared_gard = self.cv_HOST.host_run_command("PATH=/usr/local/sbin:$PATH opal-gard list", console=1)
-                self.assertIn("No GARD entries to display", cleared_gard,
-                              "Failed to clear GARD entries")
-            else: # all good so skip power cycle
-                return
+            my_term = self.cv_SYSTEM.console
+            my_pty = my_term.get_console()
+            my_term.run_command("date") # just need to run to get console setup
+            cmd_list_all = "PATH=/usr/local/sbin:$PATH opal-gard list all"
+            my_pty.sendline(cmd_list_all)
+            rc = my_pty.expect([expect_prompt, "Clear the entire GUARD", pexpect.TIMEOUT, pexpect.EOF], timeout=60)
```

Comment:
why are we looking for the confirmation string after `opal-gard list all`, that *should* just give us a list, right? The y/n prompt should come when we try to clear it, right?

Reply:
>  y/n prompt

I had an interim patch where I commented out the case to be addressed here, so it seems that with ECC errors 'opal-gard list all' we get the y/n.

```
+            #g = self.cv_HOST.host_run_command("PATH=/usr/local/sbin:$PATH opal-gard list all", console=1)
+            #if "No GARD entries to display" not in g:
+            # it seems we can hit ECC errors which require a response, so for now always clear
+            # Need to investigate having a opal-gard list all with a flag to force clearing ??
+            # PATH=/usr/local/sbin:$PATH opal-gard list all
+            # ECC: uncorrectable error: ffffffffffffffff ff
+            # The data at the GUARD partition does not appear to be valid gard data
+            # Clear the entire GUARD partition? [y/N]
```